annotations/views.py

```diff
@@ -685,16 +685,43 @@ def add_text_to_collection(request, *args, **kwargs):
     return JsonResponse({})
 
 
-@login_required
 def user_details(request, userid, *args, **kwargs):
+    """
+
+    :param request: The user request to fetch user details
+    :type: Request()
+    :param userid:
+    :type int
+    :param args:
+    :type list
+    :param kwargs:
+    :type dict
+    :return: HttpResponse
+    """
     user = get_object_or_404(VogonUser, pk=userid)
-    template = loader.get_template('annotations/user_details.html')
-    context = RequestContext(request, {
-        'user': request.user,
-        'detail_user': user,
-    })
-    return HttpResponse(template.render(context))
+    if request.user.is_authenticated():
+        template = loader.get_template('annotations/user_details.html')
+        context = RequestContext(request, {
+            'user': request.user,
+            'detail_user': user,
+        })
+    else:
+        textCount = Text.objects.filter(addedBy=user).count()
+        annotationsSet = Text.objects.filter(annotators=user).distinct()
+        annotationCount = 0
+        for text in annotationsSet:
+            annotationCount += text.annotationCount
+        textAnnotated = len(annotationsSet)
```

Comment:
Yikes! This will cause a pretty big database query. When you take the ``len()`` of a ``QuerySet``, it will evaluate the QuerySet (yielding a list of objects) first. That means that you are executing the query ``SELECT * FROM TABLE annotations_text;``, and then evaluating the length of the resulting list of objects. If you just want to get the number of records, use the QuerySet's ``count()`` method (e.g. ``annotationSet.count()``). 

Take a look at the [QuerySet documentation](https://docs.djangoproject.com/en/1.9/ref/models/querysets/).